test/system/src/test/java/io/pravega/test/system/ReadWriteAndAutoScaleWithFailoverTest.java

```diff
@@ -0,0 +1,468 @@
+                            .getReaderGroup(readerGroupName).getScope(), readerGroupManager
+                            .getReaderGroup(readerGroupName).getOnlineReaders());
+            log.info("Creating {} readers", NUM_READERS);
+            List<EventStreamReader<Long>> readerList = new ArrayList<>(NUM_READERS);
+            log.info("Scope that is seen by readers {}", scope);
+            for (int i = 0; i < NUM_READERS; i++) {
+                log.info("Starting reader: {}, with id: {}", i, readerName + i);
+                final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                        readerGroupName,
+                        new JavaSerializer<Long>(),
+                        ReaderConfig.builder().build());
+                readerList.add(reader);
+            }
+
+            // start writing asynchronously
+            List<CompletableFuture<Void>> writerFutureList = writerList.stream().map(writer ->
+                    startWriting(writer)).collect(Collectors.toList());
+
+            //start reading asynchronously
+            List<CompletableFuture<Void>> readerFutureList = readerList.stream().map(reader ->
+                    startReading(reader))
+                    .collect(Collectors.toList());
+
+            //run the failover test before scaling
+            performFailoverTest();
+
+            //bring the instances back to 3 before performing failover during scaling
+            controllerInstance.scaleService(3, true);
+            segmentStoreInstance.scaleService(3, true);
+            Thread.sleep(ZK_DEFAULT_SESSION_TIMEOUT);
+
+            //increase the number of writers to trigger scale
+            List<EventStreamWriter<Long>>  newlyAddedWriterList = new ArrayList<>(ADD_NUM_WRITERS);
+            log.info("Writers writing in the scope {}", scope);
+            for (int i = 0; i < ADD_NUM_WRITERS; i++) {
+                log.info("Starting writer{}", i);
+                final EventStreamWriter<Long> writer1 = clientFactory.createEventWriter(STREAM_NAME,
+                        new JavaSerializer<Long>(),
+                        EventWriterConfig.builder().retryAttempts(10).build());
+                newlyAddedWriterList.add(writer1);
+            }
+
+            //start writing asynchronoulsy with newly created writers
+            List<CompletableFuture<Void>> writerFutureList1 = newlyAddedWriterList.stream().map(writer ->
+                    startWriting(writer)).collect(Collectors.toList());
+
+            //run the failover test while scaling
+            performFailoverTest();
+
+            //wait for scaling
+            waitForScaling();
+
+            //bring the instances back to 3 before performing failover
+            controllerInstance.scaleService(3, true);
+            segmentStoreInstance.scaleService(3, true);
+            Thread.sleep(ZK_DEFAULT_SESSION_TIMEOUT);
```

Comment:
after performing failover test , we are scaling up the instances back to 3, before performing next failover test, and we need a sleep(zk session time out) for that.